### PR TITLE
Fix first day of the week locale issue with scheduler

### LINKED ISSUE #14312: Graphical frontend for creating/editing schedules is broken in 2022.11.1
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Only the column that has a yellow tinted backgroundbar (current day) is able to display the schedule, the other days are not displayed anymore.

This is new in this version. Has worked like a charm before. 

Very nasty bug, showing or editing schedules has become impossible

### Describe the behavior you expected

It should show the actual schedule.
And you should be able to edit it.

### Steps to reproduce the issue

1. (Try to) Create a new schedule or open an existing one.

2. When a new schedule is selected, you can only edit the local day. Other days you see what you enter but when you move your cursor away the entry is not shown anymore, although it seems to be there.

### What version of Home Assistant Core has the issue?

2022.11.1

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Google chrome Versie 107.0.5304.88

### Which operating system are you using to run this browser?

Windows11 but also on Android 12

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

I think every user will suffer from this.

### LINKED ISSUE #14354: Schedule is buggy
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

-     Somehow the Sunday schedule cannot be altered, if I add items they vanish, but when I save them I get notifications about "overlapping schedules"
-     Renaming the entity or changing its icon has no effect after its initial creation. Only restarting home assistant makes these changes take effect.

### Describe the behavior you expected

Schedules for Sunday to show up.

More info here: https://github.com/home-assistant/core/issues/81710

### Steps to reproduce the issue

1. Use schedule as described above.

### What version of Home Assistant Core has the issue?

2022.11.1

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

_No response_

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

In locales that show Monday as the first day of the week, everything was fine. However in locales that show Sunday as the week start (e.g. English US), the schedule events added for Sunday would get hidden as they were incorrectly moved a week away.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #14312 fixes #14354
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by bramkragten — APPROVED** [review]

File: src/panels/config/helpers/forms/ha-schedule-form.ts [hunk 1]
```diff
@@ -208,16 +208,22 @@ class HaScheduleForm extends LitElement {
   private get _events() {
     const events: any[] = [];
     const currentDay = new Date().getDay();
-    const baseDay = currentDay === 0 ? 7 : currentDay;
+    const baseDay =
+      currentDay === 0 && firstWeekdayIndex(this.hass.locale) === 1
+        ? 7
```
Comment (bramkragten) [inline src/panels/config/helpers/forms/ha-schedule-form.ts:213]: We should fix this better, as firstWeekDay could also be on saturday

Comment (Rudd-O) [reply src/panels/config/helpers/forms/ha-schedule-form.ts:213]: Yes this is not correct.

Comment (spacegaier) [reply src/panels/config/helpers/forms/ha-schedule-form.ts:213]: > We should fix this better, as firstWeekDay could also be on saturday

Yes, Saturday was not considered. I am trying that right now, but haven't gotten a solution yet.

Comment (spacegaier) [reply src/panels/config/helpers/forms/ha-schedule-form.ts:213]: Will be covered by https://github.com/home-assistant/frontend/pull/14566

**Pull request merged**